core/genesis.go

```diff
@@ -296,6 +300,10 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, triedb *triedb.Database, g
 			if overrides != nil && overrides.OverrideOptimismInterop != nil {
 				config.InteropTime = overrides.OverrideOptimismInterop
 			}
+			if overrides != nil && config.IsOptimism() {
```

Comment:
What is the purpose of overrides?

Reply:
After further thought, it seems `EnableSoulGasToken`/`IsSoulBackedByNative` don't have to be override-able, they can now only be configured in genesis.